xgcm/grid.py

```diff
@@ -1262,50 +1262,46 @@ def check_neighbor(link, position):
     def set_metrics(self, key, value, overwrite=False):
         metric_axes = frozenset(_maybe_promote_str_to_list(key))
         axes_not_found = [ma for ma in metric_axes if ma not in self.axes]
-
         if len(axes_not_found) > 0:
             raise KeyError(
                 f"Metric axes {axes_not_found!r} not compatible with grid axes {tuple(self.axes)!r}"
             )
 
         metric_value = _maybe_promote_str_to_list(value)
-        existing_metric_axes = set(self._metrics.keys())
+        for metric_varname in metric_value:
+            if (metric_varname not in self._ds.variables) and (
+                metric_varname not in self._ds.coords
+            ):
```

Comment:
```suggestion
            if metric_varname not in self._ds.variables:
```

that should check all variables including those in `.coords`